streetcrm/formfields.py

```diff
@@ -64,27 +64,33 @@ class TwelveHourTimeField(forms.TimeField):
 
     def to_python(self, value):
         """ Convert time to datetime.time object """
-        time, suffix = value
+        hour, minute, suffix = value
 
-        # If no time has been specified we'll get ["", ""]
-        if time in self.empty_values and suffix in self.empty_values:
+        # If no hour has been specified we'll get ["", ""]
+        if (hour in self.empty_values and
+            minute in self.empty_values and
+            suffix in self.empty_values):
             return None
 
-        # As you can't deselect radio buttons, assume no search on time is
-        # desired if we have no time but we have a suffix selected.
-        if time in self.empty_values and suffix not in self.empty_values:
+        # As you can't deselect radio buttons, assume no search on hour is
+        # desired if we have no hour but we have a suffix selected.
+        if hour in self.empty_values and suffix not in self.empty_values:
             return None
 
         # If no suffix has been supplied then we should display an error
-        if time not in self.empty_values and suffix in self.empty_values:
+        if hour not in self.empty_values and suffix in self.empty_values:
```

Comment:
We check `if hour not in self.empty_values` multiple times (this is just the first).  Since there's never a time -- hah, get it? -- when it's okay for the hour to be one the empty values, why not just check for that once, near the top, and then assume it for the rest of the method, thus simplifying subsequent conditionals?

Reply:
Ah!  Yes, great catch (and fixed).